lmdeploy/pytorch/kernels/triton_utils.py

```diff
@@ -0,0 +1,464 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import functools
+import inspect
+from typing import Callable, Dict, Sequence, Union, cast, overload
+
+import torch
+import triton
+from packaging import version
+from triton import JITFunction
+
+if version.parse(triton.__version__) <= version.parse('2.2.0'):
+
+    def get_kernel_meta(tensor: torch.Tensor):
+        """kernel meta."""
+        from triton.runtime.jit import get_cuda_stream
+
+        device = tensor.device
+        device_idx = device.index
+        device_type = device.type
+        stream = get_cuda_stream(device_idx)
+        return dict(device=device, device_type=device_type, stream=stream)
+else:
+
+    KERNEL_META = dict()
+
+    def get_kernel_meta(tensor: torch.Tensor):
+        """kernel meta."""
+        return KERNEL_META
```

Comment:
shall we put `get_kernel_meta` outside the condition and move condition inside the function?

Reply:
The function would be called on every kernel call. Move version check inside the function would enlarge kernel launch time.